apps/src/templates/courseOverview/ResourcesEditor.js

```diff
@@ -46,7 +61,13 @@ export default class ResourcesEditor extends Component {
       newResources[index].link = defaultLinks[type];
     }
 
-    this.setState({resources: newResources});
+    let errorString = '';
+    let types = newResources.map(resource => resource.type).filter(resource => resource);
```

Comment:
I think `.filter(Boolean)` actually might be a clearer way of expressing what this is doing. At the least, it should probably be `.filter(type => !!type)`